### PR TITLE
8329655: Cleanup KlassObj and klassOop names after the PermGen removal

### PR DESCRIPTION
**Pull request by stefank**
We have a few places that uses the terms `KlassObj` and `klassOop` when referring to Klasses. This is old code from before the PermGen removal, when Klasses also were Java objects.

These names tripped me up when I was reading the heap heapInspection.cpp and first though we were mixing the klass *mirror* objects and klass pointers in the hash code calculation:
```
   // An aligned reference address (typically the least
   // address in the perm gen) used for hashing klass
   // objects.
   HeapWord* _ref;
...
_ref = (HeapWord*) Universe::boolArrayKlassObj();
...
uint KlassInfoTable::hash(const Klass* p) {
  return (uint)(((uintptr_t)p - (uintptr_t)_ref) >> 2);
}
```

I propose that we rename these functions (and stop casting the Klass* to a (HeapWord*)).

Tested with serviceability/dcmd/gc/ClassHistogramTest.java but will run this through our lower tiers.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8329655](https://bugs.openjdk.org/browse/JDK-8329655): Cleanup KlassObj and klassOop names after the PermGen removal (**Enhancement** - P4)

### Reviewers
 * [Roman Kennke](https://openjdk.org/census#rkennke) (@rkennke - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18618/head:pull/18618` \
`$ git checkout pull/18618`

Update a local copy of the PR: \
`$ git checkout pull/18618` \
`$ git pull https://git.openjdk.org/jdk.git pull/18618/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18618`

View PR using the GUI difftool: \
`$ git pr show -t 18618`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18618.diff">https://git.openjdk.org/jdk/pull/18618.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18618#issuecomment-2036719414)

### CODE REVIEW

File: src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp [hunk 1]
```diff
@@ -1199,7 +1199,7 @@ void MacroAssembler::lookup_interface_method(Register recv_klass,
 
   ldrw(scan_temp, Address(recv_klass, Klass::vtable_length_offset()));
 
-  // %%% Could store the aligned, prescaled offset in the klassoop.
```
[the remark this answers is not in the document]
Comment (stefank) [reply src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp:1202]: I think it is an old-style TODO. I'm considering if we shouldn't just remove these comments. What do people think about that?

[the remark this answers is not in the document]
Comment (rkennke) [reply src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp:1202]: I'm not even sure what they want to say, really. Should be good to remove, and if anybody can make sense of it, record an issue in the bug-tracker?

[the remark this answers is not in the document]
Comment (stefank) [reply src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp:1202]: OK. I removed the %%%. I'll wait a little bit to see if someone else wants to keep them for some reason, if not, I'll remove them.

**Review by rkennke — APPROVED** [review]
Thanks! Looks good to me, now.

Roman

File: src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (coleenp) [reply src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp:1202]: I think leaving these comments without the %%% seems fine.  Describing this idea in a CR is a lot more difficult than seeing it in context as commentary, and unless the enhancement has other motivation, it won't be picked up.  Leaving the comment as a clue seems useful.

File: src/hotspot/share/memory/heapInspection.hpp [hunk 2]
```diff
@@ -108,9 +108,8 @@ class KlassInfoTable: public StackObj {
   size_t _size_of_instances_in_words;
 
   // An aligned reference address (typically the least
-  // address in the perm gen) used for hashing klass
```
Comment (coleenp) [inline src/hotspot/share/memory/heapInspection.hpp:111]: Rats I missed this.

**Review by coleenp — APPROVED** [review]
This is good. The Obj was confusing.

**Pull request closed without merging**